jurity/recommenders/intralist_diversity.py

```diff
@@ -0,0 +1,255 @@
+            parameter, or a metric listed in pairwise.PAIRWISE_DISTANCE_FUNCTIONS. If metric is a callable function,
+            it is called on each pair of instances (rows) and the resulting value recorded.
+            The callable should take two arrays from X as input and return a value indicating the distance between them.
+        num_runs: int
+            num_runs is used to report the approximation of Intra-List Diversity over multiple runs on smaller
+            samples of users, default=10, for a speed-up on evaluations. The sampling size is defined by
+            user_sample_size. The final result is averaged over the multiple runs.
+        
+        """
+
+        self.user_id_column = user_id_column
+        self.item_id_column = item_id_column
+        self.item_features = item_features
+        self.click_column = click_column
+        self.k = k
+        self.user_sample_size = user_sample_size
+        self.seed = seed
+        self.metric = metric
+        self.num_runs = num_runs
+        self._validate_arguments()
+
+    def get_score(self, predicted_results: pd.DataFrame, batch_accumulate: bool = False,
+                  return_extended_results: bool = False) -> Union[float, dict]:
+        """Evaluates the current metric on the given data.
+
+        Parameters
+        ---------
+        predicted_results: pd.DataFrame
+            A pandas DataFrame for the recommended user item interaction data, captured from a recommendation algorithm.
+            The DataFrame should contain a minimum of two columns, including self._user_id_column, self._item_id_column,
+            and anything else the metric may need. Each row contains the interaction of one user with one item, and the
+            scores associated with this interaction. There can be multiple interactions per user, and there can be
+            multiple users per DataFrame. However, the interactions for a specific user must be contained within a
+            single DataFrame.
+        batch_accumulate: bool
+            Should not be True for calculating Intra-List Diversity while it is kept for making the API design
+            consistent across different recommender metrics.
+        return_extended_results: bool
+            Whether the extended results such as the support should also be returned. If specified, the returned results
+            will be of type ``dict``. Intra-list diversity currently returns ``Intra-List Diversity`` and
+            the ``support``, which is the number of unique users to calculate it.
+
+        Returns
+        -------
+        metric: Union[float, dict]
+            The averaged result(s). The return type is determined by ``return_extended_results`` parameters.
+        """
+
+        if batch_accumulate:
+            raise ValueError("Batch_accumulate can not be set as True for Intra-List Diversity.")
```

Comment:
For intra-list, would it make sense to be able to support batch_accumulate, since the results can be calculated for each user in one batch and then the user-level results can be stacked with the following batches. Thoughts? @dorukkilitcioglu @bkleyn

Reply:
Ah, I totally missed that this could be used. Yeah, we can support batch_accumulate because the calculations are done for the items per user, without users interacting with each other.

Reply:
Yes, batch accumulate should be valid for intra-list.